packages/devtools_shared/lib/src/server/server_api.dart

```diff
@@ -17,6 +18,17 @@ import '../extensions/extension_manager.dart';
 import 'file_system.dart';
 import 'usage.dart';
 
+// TODO: confirm how to get additional parameters; switch away
+// from development constructor too
+final Analytics analytics = Analytics.development(
+  tool: DashTool.devtools,
+  dartVersion: 'TBD',
+  clientIde: 'TBD',
+  enabledFeatures: 'feature1-feature2',
+  flutterChannel: 'TBD',
+  flutterVersion: 'TBD',
+);
+
```

Comment:
Swap this out

Reply:
@kenzieschmoll any recommendations for where I should create the `Analytics` instance? Ideally it can be somewhere it can get imported from or somewhere near the entrypoint so you can direct inject it into where it is needed

Reply:
How do we get these values from other tools? 

Since this code is on the `devtools_server` where we have access to `dart:io`, one idea is to run `flutter --version` and parse the output. We have a class that parses the output from Flutter Tools' registered `flutterVersion` service here: https://github.com/flutter/devtools/blob/master/packages/devtools_app_shared/lib/src/service/flutter_version.dart/#L16. We could move this class to devtools_shared if it was useful for parsing the output from `flutter --version`.

A couple things we will need to be careful about:
- using the `flutter` that this DevTools server instance is being run from, since a user could have multiple `flutter`'s on their machine / path.
- ensuring these values can be null when `flutter` is not on the user's machine (for example if DevTools was being used from a pure Dart user who only writes Dart CLI apps or Dart web apps.

WRT for where to create the Analytics instance, where do we create the Analytics instance for other tools? Near the root of initialization? Some ideas I have:
- create it in the constructor of the DevToolsServer class: https://github.com/dart-lang/sdk/blob/main/pkg/dds/lib/devtools_server.dart#L23
- make it a late final variable that is created and passed into `defaultHandler` similar to what we do for `clientManager`: https://github.com/dart-lang/sdk/blob/main/pkg/dds/lib/devtools_server.dart#L265-L271

both of these scenarios would require that the `Analytics` instance is passed in as a parameter to `ServerApi.handle` (similar to how `extensionsManager` and `deepLinkManager` are passed in): https://github.com/flutter/devtools/blob/master/packages/devtools_shared/lib/src/server/server_api.dart/#L37

It is also worth noting that `ServerApi.handle` is called from google3 on a separate path, so we'd have to make sure we create the `Analytics` instance properly there too.

Reply:
> using the flutter that this DevTools server instance is being run from, since a user could have multiple flutter's on their machine / path.

Yeah this was the first thing I was concerned about, since devtools also downloads its own separate flutter sdk in `/devtools/tool/flutter-sdk`, wouldn't running `flutter --version --machine` return versioning information for `flutter` that is on the path?

For example, the flutter sdk that is in the devtools directory is version `"frameworkVersion": "3.20.0-0.0.pre",` while my flutter in path resolves to `"frameworkVersion": "3.20.0-2.0.pre.9",`... which of these values would we want to report for analytics?

Reply:
The `tool/flutter-sdk` Fluter isn't used in production. This is just for all of our repository management scripts as well as for the workflows we run on the CI.